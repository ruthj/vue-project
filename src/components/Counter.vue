<template>
    <h2>{{counterTitle}}</h2>
    <p>{{counter}}<sup>2</sup>={{squareCounter}}</p>
    <p data-testId="counterTest">{{counter}}</p>

    <div class='botn'>
        <button @click="increase">+</button>
        <button @click="decrease">-</button>
    </div>
</template>

<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            default: 100,

            validator(value){
                return value >= 0
            }
        }
    },

    data(){
        return{
            counter: this.start
        }
    },
    methods:{
        getSquareValue(){
            return this.counter * this.counter
        }, 
        increase(){
            this.counter++ 
            
        },
        decrease(){
            this.counter-- 
            
        }
       
    },
    computed:{
        squareCounter(){
            return this.counter * this.counter
        },
        counterTitle(){
            return this.title || 'Counter from computed' 
        }
    }
}

</script>

<style>
    button{
        background-color: #64B687;
        border-radius:5px ;
        border: 1px solid white;
        cursor: pointer;
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover{
        background-color: #5aa67b;
        transition: 0.3s ease-in-out;
    }
</style>