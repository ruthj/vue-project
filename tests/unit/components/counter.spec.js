import {shallowMount} from '@vue/test-utils'
import Counter from '@/components/Counter'


describe('Counter component', () => {

    let wrapper 

    beforeEach(() => {
        wrapper = shallowMount(Counter); //mounts the component no the app.js
    })


    test('Should match with snapshot', () => {
        const wrapper = shallowMount(Counter)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('Should h2 has default value', () => {
        //const wrapper = shallowMount(Counter)
        expect(wrapper.find('h2').exists()).toBeTruthy()
        const h2Vule = wrapper.find('h2').text()
        //console.log(h2.text())
        expect(h2Vule).toBe('Counter from computed')
    })

    test('Should be default 100', () => {
        //const wrapper = shallowMount(Counter)
        const pTags = wrapper.find('[data-testId="counterTest"]').text()
        //expect(pTags[1].text()).toBe("100")
        expect(pTags).toBe('100')

    })

    test('Increment and decrement by 1', async() => {
        //const wrapper = shallowMount(Counter)
        const [increaseBtn, decreseBtn] = wrapper.findAll('button')
        
        await increaseBtn.trigger('click')
        await increaseBtn.trigger('click')
        await increaseBtn.trigger('click')
        await decreseBtn.trigger('click')
        await decreseBtn.trigger('click')
        
        const value = wrapper.find('[data-testId="counterTest"]').text()
        expect(value).toBe('101')

    })

    test('Test Props', () => {
       
        const {start} = wrapper.props()
        const value = wrapper.find('[data-testId="counterTest"]').text()
        expect(Number(value)).toBe(start)
    })

    test('Property title read and show ', () => {
        const title = 'Hi world!!!!!'
       const wrapper = shallowMount(Counter,{
           props:{
               title
           }
       })
       expect(wrapper.find('h2').text()).toBe(title)
       
     })

})