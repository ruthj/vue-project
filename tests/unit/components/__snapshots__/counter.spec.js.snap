// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Should match with snapshot 1`] = `
<h2>Counter from computed</h2>
<p>100<sup>2</sup>=10000</p>
<p data-testid="counterTest">100</p>
<div class="botn"><button>+</button><button>-</button></div>
`;
